local_server/telemetry_server/server.py

```
@@ -0,0 +1,405 @@
+    sorted_days = sorted(stats.get("daily_totals", {}).items(), reverse=True)
+
+    # --- DAU 趋势（最近的放右边） ---
+    dau_trend = user_metrics.get("dau_trend", {})
+    dau_sorted = sorted(dau_trend.items())  # 日期升序
+    dau_labels = [d[5:] for d, _ in dau_sorted]  # MM-DD
+    dau_values = [v for _, v in dau_sorted]
+    dau_max = max(dau_values) if dau_values else 1
+
+    # DAU 柱状图（纯 CSS，无 JS 依赖）
+    dau_bars = ""
+    for i, (label, val) in enumerate(zip(dau_labels, dau_values)):
```

Ruff (0.15.5)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)